onedrivesdk/src/main/java/com/onedrive/sdk/http/DefaultHttpProvider.java

```diff
@@ -278,6 +299,16 @@ public void run() {
                     return result;
                 }
 
+                if (isUploadSession(resultClass)) {
+                    mLogger.logDebug("Handling upload session completed");
+                    in = new BufferedInputStream(connection.getInputStream());
+                    if (connection.getResponseCode() == httpCreatedResponseCode) {
+                        Item item = this.handleJsonResponse(in, Item.class);
```

Comment:
could you please detail at it? if we change the name in $metadata, then the templates need regenerated with the new name and I can foresee the build of SDK will fail because they cannot find the old one if the \generated updated correctly. But at some point you are right, probably BaseItem.class is better?